aten/src/ATen/native/cuda/MultinomialKernel.cu

```diff
@@ -201,21 +199,22 @@ sampleMultinomialOnce(int64_t* dest,
     }
 
     // threadIdx.x == 0 has the sum value from this
-    sum = reduceBlock(asmem, blockDim.x, sum, ReduceAdd<accscalar_t>(), accZero);
+    sum = reduceBlock(smem, blockDim.x, sum, ReduceAdd<accscalar_t>(), accZero);
```

Comment:
consider using `block_reduce` from native/cuda/block_reduce.cuh here instead of legacy reduceBlock